client/js/genre.js

```diff
@@ -1,24 +1,22 @@
 /**
- * fetches youtube api calls from YoutubeServlet.java
+ * @file fetches youtube api calls from YoutubeServlet.java
  * and displays on youtube-genre.html
  */
 
-const genreBlock = document.getElementById('genres');
-
 /**
- * fetches genre count hashmap from /api/youtube and updates html
+ * fetches and returns genre analysis object from /api/youtube
+ *
+ * @returns {Promise<object>} An obj containing Youtube genre data and stats
  */
-async function displayMusicGenre() {
-  // keep track of num_videos in URL w/o reload
-  history.pushState('', '', `youtube-genre.html`);
-
-  const response = await fetch(`/api/youtube`);
+async function fetchMusicGenre() {
+  const response = await fetch('/api/youtube');
   if (response.status == 401) {
     // no oauth login so redirect to new page
     window.open('/api/oauth/login/youtube');
   }
 
   const genreCount = await response.text();
-  genreBlock.innerHTML = genreCount;
+  return JSON.parse(genreCount);
 }
 
+export const GENRE_ANALYSIS_PROMISE = fetchMusicGenre();
```

Comment:
You shouldn't export a call to a promise. This line will trigger this fetch /anytime/ this file is imported. This may be what you want in the short term, but isn't what folks will expect from a library/file they import. You should instead export `fetchMusicGenre`, and call it in your other file.

Reply:
In general, libraries and imported files shouldn't have side effects. They should be idempotent.

Reply:
oh that is right